test/e2e/operator_test.go

```diff
@@ -101,6 +122,72 @@ func TestDefaultClusterIngressExists(t *testing.T) {
 	}
 }
 
+func TestClusterIngressControllerCreateDelete(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ci := &ingressv1alpha1.ClusterIngress{}
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: ns, Name: "default"}, ci); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to get default IngressController: %v", err)
+	}
+
+	dnsConfig := &configv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, dnsConfig); err != nil {
+		t.Fatalf("failed to get DNS 'cluster': %v", err)
+	}
+
+	ing := newIngressController(ingressControllerName, ns, ingressControllerName+"."+dnsConfig.Spec.BaseDomain)
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		if err := cl.Create(context.TODO(), ing); err != nil {
+			if errors.IsAlreadyExists(err) {
```

Comment:
Not sure what we gain by polling here; if the instance already exists, we don't know where it came from and shouldn't test it... what about just calling Create and failing if any error is returned?

Reply:
Good point. I'll update.

Reply:
I think your point can be applied to `IsAlreadyExists for the delete operation too.